src/components/TodoFilter/TodoFilter.tsx

```diff
@@ -1,34 +1,77 @@
-export const TodoFilter = () => (
-  <form className="field has-addons">
-    <p className="control">
-      <span className="select">
-        <select data-cy="statusSelect">
-          <option value="all">All</option>
-          <option value="active">Active</option>
-          <option value="completed">Completed</option>
-        </select>
-      </span>
-    </p>
-
-    <p className="control is-expanded has-icons-left has-icons-right">
-      <input
-        data-cy="searchInput"
-        type="text"
-        className="input"
-        placeholder="Search..."
-      />
-      <span className="icon is-left">
-        <i className="fas fa-magnifying-glass" />
-      </span>
-
-      <span className="icon is-right" style={{ pointerEvents: 'all' }}>
-        {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-        <button
-          data-cy="clearSearchButton"
-          type="button"
-          className="delete"
+import React, {
+  ChangeEvent,
+  useContext,
+  useState,
+} from 'react';
+import { Select } from '../../types/Select';
+import { TodosContext } from '../TodosProvider';
+
+export const TodoFilter: React.FC = React.memo(() => {
+  const [query, setQuery] = useState('');
+  const [selectedValue, setSelectedValue] = useState('all');
+
+  const {
+    FiltredTodosBySearch,
+    FiltredTodosBySelect,
+  } = useContext(TodosContext);
+
+  const changeQuery = (event: ChangeEvent<HTMLInputElement>) => {
+    setQuery(event.target.value);
+    FiltredTodosBySearch(event.target.value);
+  };
+
+  const clearQuery = () => {
+    setQuery('');
+    FiltredTodosBySearch('');
+  };
+
+  const changeSelectedValue = (event: ChangeEvent<HTMLSelectElement>) => {
+    setSelectedValue(event.target.value);
+    FiltredTodosBySelect(event.target.value);
```

Comment:
can you understand from the variable name that it's a function? let's think more about naming 🐝